tests/tests_integration/test_api/test_relationships.py

```diff
@@ -100,7 +103,7 @@ class TestRelationshipsAPI:
     def test_get_single(self, new_relationship):
         res = API_REL.retrieve(external_id=new_relationship.external_id)
         assert isinstance(res, Relationship)
-        assert res.dump()["confidence"] == 0.9
+        assert res.dump()["confidence"] == 1
```

Comment:
This was a failing edge case a few days ago in data viz team.

Reply:
what was wrong? floating point roundoff ?

Reply:
this edge case never failed for the retrieve endpoint - it would fail for the filter endpoint (for the curious, the confidence filter would run on all filter requests, and the default values were set to the extreme allowed values for confidence, and since the upper limit was also inclusive, every single filter request would run 0 <= confidence < 1 - big oopsie)

Reply:
I've added the test for that in filter